### PR TITLE
Support old-style jump labels

### PR DESCRIPTION
**Pull request by Claes65**

### CODE REVIEW

**Comment by MarkusAmshove** [comment]
Seems fine to me, not adding/deleting any diagnostics on our side. If you're ready I can merge :)

**Comment by Claes65** [comment]
I best compare before you merge this. ;)

**Comment by MarkusAmshove** [comment]
You might need to merge main into this, as i've added another build step :)

File: libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java [hunk 1]
```diff
@@ -843,6 +850,33 @@ private void consumeAsteriskOrSystemVariable()
...
+	private boolean consumeNumberedLabel()
+	{
+		var previous = previous();
+		if (previous != null && previous.kind() == SyntaxKind.IDENTIFIER)
+		{
+			return false;
+		}
+
+		var i = 1; // Skip '(', then check if next 4 are digits
+		while (!scanner.isAtEnd() && Character.isDigit(scanner.peek(i)) && i < 5)
```
Comment (MarkusAmshove) [inline libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java:862]: This should be `scanner.isAtEnd(i)` to check if the offset `i` is still in bounds

**Review by MarkusAmshove — CHANGES_REQUESTED** [review]

File: libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java (hunk 1 above)
Comment (Claes65) [reply libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java:862]: good find 👍

**Review by MarkusAmshove — APPROVED** [review]

**Pull request merged**